WordPress/src/androidTest/java/org/wordpress/android/e2e/EditorTests.java

```diff
@@ -1,29 +1,69 @@
 package org.wordpress.android.e2e;
 
+import android.Manifest.permission;
 import android.support.test.rule.ActivityTestRule;
+import android.support.test.rule.GrantPermissionRule;
 
 import org.junit.Rule;
 import org.junit.Test;
+import org.wordpress.android.R;
 import org.wordpress.android.e2e.components.MasterbarComponent;
 import org.wordpress.android.e2e.pages.EditorPage;
 import org.wordpress.android.e2e.pages.MySitesPage;
 import org.wordpress.android.support.BaseTest;
 import org.wordpress.android.ui.WPLaunchActivity;
 
+import java.time.Instant;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.pressBack;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
 import static junit.framework.TestCase.assertTrue;
 import static org.wordpress.android.BuildConfig.E2E_WP_COM_USER_SITE_ADDRESS;
+import static org.wordpress.android.support.WPSupportUtils.checkViewHasText;
 import static org.wordpress.android.support.WPSupportUtils.sleep;
+import static org.wordpress.android.support.WPSupportUtils.waitForElementToNotBeDisplayed;
 
 public class EditorTests extends BaseTest {
     @Rule
     public ActivityTestRule<WPLaunchActivity> mActivityTestRule = new ActivityTestRule<>(WPLaunchActivity.class);
 
+    @Rule
+    public GrantPermissionRule mRuntimeImageAccessRule = GrantPermissionRule.grant(permission.WRITE_EXTERNAL_STORAGE);
+
+    @Test
+    public void testPublishSimplePost() {
+        String title = "Hello Espresso!";
+        String content = "Lorem ipsum dolor sit amet, consectetur adipiscing elit.";
+
+        loginIfNecessary();
+
+        MasterbarComponent mb = new MasterbarComponent().goToMySitesTab();
+        sleep();
+        mb.clickBlogPosts();
+
+        new MySitesPage()
+                .startNewPost(E2E_WP_COM_USER_SITE_ADDRESS);
+
+        EditorPage editorPage = new EditorPage();
+        editorPage.enterTitle(title);
+        editorPage.enterContent(content);
+        boolean isPublished = editorPage.publishPost();
+        assertTrue(isPublished);
+    }
+
     @Test
-    public void testPublishPost() {
-        String title = "Title";
-        String content = "Content";
+    public void testPublishFullPost() {
+        String title = "Hello Espresso!";
+        String content = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod "
+                         + "tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud "
+                         + "exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.";
+        String category = "Android Test";
+        long now = Instant.now().toEpochMilli();
+        String tag = "Tag " + now;
 
-        wpLogin();
+        loginIfNecessary();
```

Comment:
Would it be safer for this to be:

```
logoutIfNecessary();
wpLogin();
```

The `loginIfNecessary()` method does nothing to verify we are logged in as the correct user. With mocking, this is usually the same user so may not matter. But this could change. What do you think?

Reply:
I was thinking about this recently but didn't give it much thought since these will run in a controlled environment, for the most part, I think it's a similar case in the iOS tests where if it's already logged in(no matter who the user is) the tests continue... cc @rachelmcr 

I think the change you suggested is a good way to cover this and prevent some against unnecessary errors.

Reply:
Agreed, I've had the same thought recently about verifying we are logged in as the correct user. I started to do this in the login improvements I was working on for iOS but hadn't added it to all login tests and flows. Let's do it.

I opened an issue to track this on the iOS side: https://github.com/wordpress-mobile/WordPress-iOS/issues/11759